docs/io.cozy.rules.md

```diff
@@ -0,0 +1,37 @@
+[Table of contents](README.md#table-of-contents)
+
+# Cozy Rules doctype
+
+## `io.cozy.rules`
+
+This doctype represents rules, used to target documents in the Cozy.
```

Comment:
I am bit hesitant on the vocabulary here. 

- "rules" : It seems to me that this is like a "query definition" in cozy-client parlance. Should we use the same vocabulary ? Are they the same concept or is there a subtlety that I missed ? 

- "target" here is a bit unclearer, "retrieve", "fetch" or "select" seems more appropriate to me as they are used more often in the context of data retrieval.

Reply:
* rules: it is the vocabulary taken from cozy-procedure: https://github.com/cozy/cozy-libs/blob/6af091bc50193edb9c2dbc6a88e00631472436d4/packages/cozy-procedures/src/templates/creditApplicationTemplate.js#L182-L191 , I think changing by the more technical "query definition"  would introduce unecessary complexity for the reader. 
* "target": ok, I'm a bit surprised that's unclear at first read, but retrieve is fine also. (fetch seems too network-related and select too sql-oriented).

Reply:
> it is the vocabulary taken from cozy-procedure

So ?

> by the more technical "query definition"

I think it's expected that in a technical documentation, we refer to technical things.  I really have a hard time with "rules", it is very broad, and it does not convey the fact that rules are used to query documents. At first I thought "rules" in the context of a todo application, would be an "order" in which to do the tasks, or whether a task could be done given a context.

Also, it seems like we have something that is really close to query definitions, a concept that we already have, and I think what would help the reader would be that the number of concepts inside Cozy is not too large. If we have the opportunity to reuse a concept, I think we ought to do it.

> select too sql-oriented

mmh, I don't think so, we have `selectors` in CouchDB :)

Reply:
> So ?

I mentioned cozy-procedure because the objective would eventually to store those rules in couchdb, so we could extend this notion of matching rules elsewhere. 

> I really have a hard time with "rules", it is very broad, and it does not convey the fact that rules are used to query documents. 

I see your point, but it might be biaised because you saw it in the context of todos, which is not meant to be the sole use-case.

>I think it's expected that in a technical documentation, we refer to technical things.

Indeed, but it's not expected to master all the concepts, and it seems to me that "query definition" is a really specific terminology less understood than the more generic "rules". More importantly, a rule could indeed refer to a query definition but also to something else: imagine a rule used to recognize faces on a picture: the rule could be a set of parameters used by some ML treatment. 
So, for me, a rule has a broader semantic than just a query definition.

> mmh, I don't think so, we have selectors in CouchDB :)

Fair point, but in my mind, "to select" data is not completely tied to "a selector", which is purely subjective :) Anyway, "retrieve" is fine.

Reply:
> More importantly, a rule could indeed refer to a query definition but also to something else: imagine a rule used to recognize faces on a picture: the rule could be a set of parameters used by some ML treatment.

I am not sure I understand, do you suggest that the io.cozy.rules doctype could one day store the set of hyperparameters of a ML model ?

Reply:
Done in 4c57a59